controllers/cluster_controller.go

```diff
@@ -257,14 +258,15 @@ func (r *ClusterReconciler) listChildren(ctx context.Context, cluster *clusterv1
 		return nil
 	}
 
-	lists := map[string]runtime.Object{
-		"MachineDeployment": machineDeployments,
-		"MachineSet":        machineSets,
-		"Machine":           machines,
+	lists := []runtime.Object{
+		machineDeployments,
+		machineSets,
+		machines,
+		controlplanes,
 	}
-	for name, list := range lists {
+	for _, list := range lists {
 		if err := meta.EachListItem(list, eachFunc); err != nil {
-			return nil, errors.Wrapf(err, "error finding %s children of cluster %s/%s", name, cluster.Namespace, cluster.Name)
```

Comment:
I'd prefer not to lose the metadata about which type we're processing. You could keep the map + add a list for ordering. Range over the list, pull items from the map.

Reply:
I don't think that's actually needed, the only error we return (and the ones returned from EachListItem) already contain the type information. This seems really extra

Reply:
Fair enough